research/cifar10/evaluate_on_test.py

```diff
@@ -0,0 +1,748 @@
+                )
+                test_results[f"agg_client_{client_number}_post_best_model_local_avg"] = avg_test_metric
+                test_results[f"agg_client_{client_number}_post_best_model_local_std"] = std_test_metric
+
+        if eval_last_post_aggregation_local_models:
+            avg_test_metric, std_test_metric = get_metric_avg_std(post_last_local_test_metrics)
+            log(
+                INFO,
+                f"""Client {client_number} Post-aggregation Last Model Average Test
+                  Performance on own Data: {avg_test_metric}""",
+            )
+            log(
+                INFO,
+                f"""Client {client_number} Post-aggregation Last Model St. Dev. Test
+                  Performance on own Data: {std_test_metric}""",
+            )
+            test_results[f"client_{client_number}_post_last_model_local_avg"] = avg_test_metric
+            test_results[f"client_{client_number}_post_last_model_local_std"] = std_test_metric
+
+            if eval_over_aggregated_test_data:
+                avg_test_metric, std_test_metric = get_metric_avg_std(post_last_local_agg_test_metrics)
+                log(
+                    INFO,
+                    f"""Aggregated Client {client_number} Post-aggregation Last Model Average Test
+                    Performance on own Data: {avg_test_metric}""",
+                )
+                log(
+                    INFO,
+                    f"""Aggregated Client {client_number} Post-aggregation Last Model St. Dev. Test
+                    Performance on own Data: {std_test_metric}""",
+                )
+                test_results[f"agg_client_{client_number}_post_last_model_local_avg"] = avg_test_metric
+                test_results[f"agg_client_{client_number}_post_last_model_local_std"] = std_test_metric
+
+        if eval_best_global_model:
+            avg_server_test_global_metric, std_server_test_global_metric = get_metric_avg_std(best_server_test_metrics)
+            log(
+                INFO,
+                f"Server Best model Average Test Performance on Client {client_number} "
+                f"Data: {avg_server_test_global_metric}",
+            )
+            log(
+                INFO,
+                f"Server Best model St. Dev. Test Performance on Client {client_number} "
+                f"Data: {std_server_test_global_metric}",
+            )
+            test_results[f"server_best_model_client_{client_number}_avg"] = avg_server_test_global_metric
+            test_results[f"server_best_model_client_{client_number}_std"] = std_server_test_global_metric
+
+            if eval_over_aggregated_test_data:
+                avg_server_test_global_metric, std_server_test_global_metric = get_metric_avg_std(
+                    best_server_agg_test_metrics
+                )
+                log(
+                    INFO,
+                    f"Aggregated Server Best model Average Test Performance on Client {client_number} "
```

Comment:
I may be missing something here, but this value will be the server models average performance on the entire test set no just Client X's data?

Reply:
Yeah all bad wordings, to distinguish I basically added Aggregated on each sentence, but now that you point out I can see it does not making sense.